src/org/ods/orchestration/phases/DeployOdsComponent.groovy

```diff
@@ -58,21 +60,29 @@ class DeployOdsComponent {
                     applyTemplates(openShiftDir, deploymentMean)
 
                     def retries = project.environmentConfig?.openshiftRolloutTimeoutRetries ?: 10
-                    def podData = null
+                    def podDataContext = [
```

Comment:
If this data is only to generate an error message, please, generate only when you are about to echo the message.

Reply:
May rationale here was that the case where no pods are not found is not typically happening. If we move the message generation in a conditional it could be that the code itself is flawed and potentially cause an exception. 
Given that I already have tested the code as is I am now confident that the code would work and we could still move it.